src/lib.rs

```diff
@@ -76,3 +76,9 @@ fn copy_to_libc(s: &[u8]) -> *mut libc::c_char {
         string
     }
 }
+
+/// Free memory allocated by zenoh-c functions like `z_keyexpr_to_string`, `zc_config_get`, etc.
+#[no_mangle]
+pub unsafe extern "C" fn zc_free(p: *mut libc::c_void) {
```

Comment:
I do not see the need to have it as a Zenoh-C specific.
Zenoh-Pico already wraps `z_free` according to the allocation functions of each platform. So, as of today the user already must use `z_free` for anything that has been allocated by Zenoh. When Zenoh-Pico has its own allocation functions this rule must still remains.

Reply:
We already have clean functions for the different Zenoh types.
`char*` (e.g. when converting from a `keyexpr` to a string format) might be the one that will create issues if the user tries to release it with a plain `free`. Note, that it might be that we cannot override `free` for several reasons.

Reply:
Just discussed it with @cguimaraes. Maybe we need to return some owned string type (`z_owned_str_t` ?) and allow user to drop it with existing z_drop. This is more error-prone than require the user to remember, that some 'char*' were allocated by zenoh so it have to free it by zenoh too.
Since this is a change of existing API, not just adding new function, this need to be discussed.